File: app/src/main/java/com/dertyp7214/appstore/settings/SettingsCheckBox.java
/*
 * Copyright (c) 2018.
 * Created by Josua Lengwenath
 */

package com.dertyp7214.appstore.settings;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;

public class SettingsCheckBox extends Settings {

    private boolean checked;

    public SettingsCheckBox(String name, String text, Context context, boolean checked) {
        super(name, text, context);
        this.checked=checked;
        loadSetting();
    }

    public boolean isChecked(){
        return this.checked;
    }

    public void setChecked(boolean checked){
        this.checked=checked;
        saveSetting();
    }

    @Override
    public void saveSetting(){
        SharedPreferences preferences = context.getSharedPreferences("settings", Context.MODE_PRIVATE);
        @SuppressLint("CommitPrefEdits")
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean(name, checked);
        editor.apply();
    }

    @Override
    public void loadSetting(){
        SharedPreferences preferences = context.getSharedPreferences("settings", Context.MODE_PRIVATE);
        this.checked=preferences.getBoolean(name, checked);
    }
}

File: app/src/main/java/com/dertyp7214/appstore/settings/SettingsCheckBox.kt
/*
 * Copyright (c) 2018.
 * Created by Josua Lengwenath
 */

package com.dertyp7214.appstore.settings

import android.annotation.SuppressLint
import android.content.Context
import com.dertyp7214.appstore.Utils.Companion.getSettings

class SettingsCheckBox(name: String, text: String, context: Context, private var checked: Boolean) : Settings(name, text, context) {

    var isChecked: Boolean
        get() = this.checked
        set(checked) {
            this.checked = checked
            saveSetting()
        }

    init {
        loadSetting()
    }

    override fun saveSetting() {
        val preferences = getSettings(context)
        @SuppressLint("CommitPrefEdits")
        val editor = preferences.edit()
        editor.putBoolean(name, checked)
        editor.apply()
    }

    override fun loadSetting() {
        val preferences = getSettings(context)
        this.checked = preferences.getBoolean(name, checked)
    }
}
